doc/nrf/libraries/bootloader/bl_monotonic_counters.rst

```diff
@@ -0,0 +1,47 @@
+.. _doc_bl_monotonic_counters:
+
+Monotonic counters
+##################
+
+.. contents::
+   :local:
+   :depth: 2
+
+The monotonic counters library implements the logic for readind and writing uint16_t counters which can only increase.
+
+The library is used by :ref:`bootloader` to keep track of the version of the application.
+The monotonic counters are stored as part of the bootloader storage :ref:`doc_bl_storage`.
+When updating the application, the :ref:`doc_bl_validation` library checks if the new version is newer than the current version and only accepts the new image if it is.
+After updating, the application must then store the new version information in the bootloader storage.
+
+This functionality is implemented as a monotonic version counter that contains a series of 16-bit integer values.
+Each update to the counter is written to the next available slot.
+When reading the counter, the bootloader storage library iterates over each slot and returns the largest value.
+
+The number of available slots, thus the number of different version numbers that can be stored, is configurable through :kconfig:option:`CONFIG_SB_NUM_VER_COUNTER_SLOTS`.
```

Comment:
```suggestion
```

We can mention this efficiently above : https://github.com/nrfconnect/sdk-nrf/pull/9196/files#r1021664447